### PR TITLE
better init advi

### PR DESCRIPTION
**Pull request by ferrine**
I've changed default choice of optimizer in init_nuts as suggested by @junpenglao

### CODE REVIEW

**Comment by junpenglao** [comment]
LGTM. Tested locally on my problem and it converges quite quickly. Although it still not stops when it converge.

File: pymc3/variational/inference.py [hunk 1]
```diff
@@ -618,3 +618,7 @@ def fit(n=10000, local_rv=None, method='advi', model=None, random_seed=None, sta
                         'or Inference instance' %
                         set(_select.keys()))
     return inference.fit(n, **kwargs)
+
+def _infmean(input_array):
+    """Return the mean of the finite values of the array"""
+    return np.mean(np.asarray(input_array)[np.isfinite(input_array)])
```
Comment (fonnesbeck) [inline pymc3/variational/inference.py]: Need new line

File: pymc3/variational/inference.py [hunk 2]
```diff
@@ -618,3 +618,7 @@ def fit(n=10000, local_rv=None, method='advi', model=None, random_seed=None, sta
                         'or Inference instance' %
                         set(_select.keys()))
     return inference.fit(n, **kwargs)
+
+def _infmean(input_array):
```
Comment (fonnesbeck) [inline pymc3/variational/inference.py]: Why the underscore naming convention?

Comment (ferrine) [reply pymc3/variational/inference.py]: Suppose to use only locally in module. Maybe it's better to move inside local namespace

**Comment by ferrine** [comment]
I'll change callbacks soon, @junpenglao . I have an idea to check convergence in both relative and absolute  scale

**Comment by ferrine** [comment]
The main question is acceptable tolerance. Do we really need accurate solution? CC @fonnesbeck

**Comment by twiecki** [comment]
I think this looks good, having both convergence checks should catch the cases we care about. And the solution does not need to be exact.

**Comment by junpenglao** [comment]
@ferrine should we change the default `every` here https://github.com/pymc-devs/pymc3/blob/master/pymc3/variational/callbacks.py#L53 to lower value? It might work better with `every=100`.

**Comment by ferrine** [comment]
I think that's ready. So I merge

**Pull request merged**